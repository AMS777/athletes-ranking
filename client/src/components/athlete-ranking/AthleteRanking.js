import React, { Component } from 'react';
import { Link } from '@reach/router';

import AthletesLoader from './AthletesLoader';
import Modal from '../modal/Modal';

class AthleteRanking extends Component {
  sportSpecificData = {
    'Muay Thai': athlete => <p>KOs: {athlete.kos}</p>,
    Athletics: athlete => <p>Olimpic Records: {athlete.olimpicRecords}</p>,
    Football: athlete => <p>Goals: {athlete.goals}</p>,
  };

  constructor(props) {
    super(props);
    this.state = {
      selectedAthlete: null,
      showModal: false,
    };
  }

  toggleModal = athlete =>
    this.setState({
      selectedAthlete: athlete,
      showModal: !this.state.showModal,
    });

  getSportSpecificData = athlete => {
    return this.sportSpecificData[athlete.sport](athlete);
  };

  render() {
    const { selectedAthlete, showModal } = this.state;

    const modal = showModal ? (
      <Modal>
        <h4>Athlete</h4>
        <p>Name: {selectedAthlete.name}</p>
        <p>Sport: {selectedAthlete.sport}</p>
        {this.getSportSpecificData(selectedAthlete)}
        <p>Prestige points: {selectedAthlete.prestigePoints}</p>
        <div>
          <button
            onClick={this.toggleModal}
            className="btn waves-effect waves-light black"
          >
            Close
          </button>
        </div>
      </Modal>
    ) : null;

    return (
      <>
        <h1 style={{ fontSize: '2em' }}>Athletes Ranking</h1>

        <AthletesLoader>
          {({ athletes, loading }) => {
            if (loading) {
              return <p>Loading...</p>;
            }
            if (!athletes.length) {
              return <p>Athlete ranking empty.</p>;
            }

            return (
              <>
                <small>(Click on the athletes to view the details)</small>
                <ol
                  style={{
                    paddingLeft: '0',
                    display: 'flex',
                    flexDirection: 'column',
                    alignItems: 'center',
                  }}
                >
                  {athletes
                    .sort((a, b) => b.prestigePoints - a.prestigePoints)
                    .map((athlete, index) => (
                      // add the index to the key to avoid duplicate names
                      <li
                        key={athlete.name + index}
                        onClick={() => this.toggleModal(athlete)}
                        style={{ cursor: 'pointer' }}
                      >
                        {athlete.name} ({athlete.prestigePoints} pps)
                      </li>
                    ))}
                </ol>
              </>
            );
          }}
        </AthletesLoader>
        <Link to="athlete-form" className="btn waves-effect waves-light black">
          New athlete
        </Link>
        {modal}
      </>
    );
  }
}

export default AthleteRanking;
