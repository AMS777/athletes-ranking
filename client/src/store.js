import { createStore, applyMiddleware } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension/developmentOnly';
import reduxThunk from 'redux-thunk';
import reducers from './reducers';

const initialState = {};

const middleware = [reduxThunk];

const store = createStore(
  reducers,
  initialState,
  composeWithDevTools(applyMiddleware(...middleware)),
);

export default store;
