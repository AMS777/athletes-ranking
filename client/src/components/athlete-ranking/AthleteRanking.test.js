import React from 'react';
import { create } from 'react-test-renderer';
import { Provider } from 'react-redux';

import store from '../../store';

import AthleteRanking from './AthleteRanking';

test('snapshot', () => {
  const component = create(
    <Provider store={store}>
      <AthleteRanking />
    </Provider>,
  );
  expect(component.toJSON()).toMatchSnapshot();
});
