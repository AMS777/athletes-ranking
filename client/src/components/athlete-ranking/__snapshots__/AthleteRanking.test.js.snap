// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
Array [
  <h1
    style={
      Object {
        "fontSize": "2em",
      }
    }
  >
    Athletes Ranking
  </h1>,
  <p>
    Loading...
  </p>,
  <a
    className="btn waves-effect waves-light black"
    href="/athlete-form"
    onClick={[Function]}
  >
    New athlete
  </a>,
]
`;
